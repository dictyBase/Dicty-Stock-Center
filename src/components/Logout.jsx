import { connect } from "react-redux"
import { Redirect } from "react-router-dom"
import React, { Component } from "react"
import { logoutUser } from "actions/auth"

class Logout extends Component {
  displayName = "logout confirmation"

  componentWillMount() {
    this.props.logoutUser()
  }

  render() {
    return (
      <Redirect to="/" />
    )
  }
}

export default connect(null, { logoutUser })(Logout)
