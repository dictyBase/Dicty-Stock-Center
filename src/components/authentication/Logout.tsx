import React, { useEffect } from "react"
import { useMutation } from "@apollo/react-hooks"
import { Redirect } from "react-router-dom"
import { useAuthStore, ActionType } from "components/authentication/AuthStore"
import { LOGOUT } from "graphql/mutations"

/**
 * Allows the user to logout
 */

const Logout = () => {
  const [logout] = useMutation(LOGOUT)
  const [, dispatch] = useAuthStore()

  useEffect(() => {
    logout()
    dispatch({
      type: ActionType.LOGOUT,
    })
  }, [dispatch, logout])

  return <Redirect to="/" />
}

export default Logout
