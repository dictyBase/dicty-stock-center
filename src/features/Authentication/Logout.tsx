import React, { useEffect } from "react"
import { useMutation, useApolloClient } from "@apollo/client"
import { Redirect } from "react-router-dom"
import { useAuthStore, ActionType } from "features/Authentication/AuthStore"
import { LOGOUT } from "common/graphql/mutations"

/**
 * Logout handles the user logout process.
 */
const Logout = () => {
  const client = useApolloClient()
  const [{ token }, dispatch] = useAuthStore()
  const [logout] = useMutation(LOGOUT, {
    context: {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    },
  })

  useEffect(() => {
    logout()
    dispatch({
      type: ActionType.LOGOUT,
    })
    client.resetStore()
  }, [dispatch, logout, client])

  return <Redirect to="/" />
}

export default Logout
