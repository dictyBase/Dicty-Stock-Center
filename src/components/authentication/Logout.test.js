import React from "react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { Logout } from "./Logout"

const logoutUser = () => {}

test("matching a snapshot of Logout", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Logout logoutUser={logoutUser} />
    </BrowserRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
