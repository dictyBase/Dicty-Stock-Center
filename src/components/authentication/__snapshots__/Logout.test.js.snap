// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Logout 1`] = `null`;
